xarray/core/variable.py

```diff
@@ -57,10 +59,20 @@
     maybe_coerce_to_str,
 )
 
-NON_NUMPY_SUPPORTED_ARRAY_TYPES = (
-    indexing.ExplicitlyIndexed,
-    pd.Index,
-)
+
+def _get_NON_NUMPY_SUPPORTED_ARRAY_TYPES():
```

Comment:
This should be cached, e.g., with `functools.cache`

Also the name should be fixed to use `snake_case` (`ALL_CAPS` is for constants)